mail/rspamd/Portfile

```diff
@@ -33,49 +30,80 @@ compiler.whitelist clang macports-clang-8.0 macports-clang-7.0 macports-clang-6.
 
 depends_build-append \
                     port:pkgconfig \
+                    port:ragel \
                     port:luajit \
-                    port:ragel
+                    port:gd2 \
+                    port:fann \
+                    port:libstemmer
 
 depends_lib-append  port:libevent \
                     port:libmagic \
                     port:sqlite3 \
                     port:openssl \
-                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
-                    path:lib/libopenblas.dylib:OpenBLAS \
                     port:gmime \
                     port:lua \
-                    port:pcre \
                     port:perl5 \
                     port:redis \
                     port:hiredis \
-                    port:zlib
+                    port:zlib \
+                    path:lib/pkgconfig/glib-2.0.pc:glib2 \
+                    path:lib/libopenblas.dylib:OpenBLAS
+
+# Critical dependency note: port:pcre and port:pcre2 break the rspamd binary;
+# use native /usr/lib/libpcre.dylib.  See https://github.com/rspamd/rspamd/issues/2884
+# Do not use this the library provided by port:pcre without pcre jit debugging
+# Do not use -DENABLE_PCRE2=ON without pcre jit debugging
+# # depends_build-append  port:pcre
+# # depends_build-append  port:pcre2
 
 set rspamd_user     _rspamd
 set rspamd_group    ${rspamd_user}
 add_users ${rspamd_user} group=${rspamd_group} realname=Rspamd
 
 libpath             ${prefix}/var/lib/${name}
 
+# -DCMAKE_OSX_ARCHITECTURES="x86_64" causes the "Symbol not found: _blake2b_blocks_x86" error
+# See https://github.com/rspamd/rspamd/issues/2883
+cmake.set_osx_architectures no
+
+# Reference: https://github.com/rspamd/rspamd/blob/master/CMakeLists.txt
+# Also see https://github.com/rspamd/rspamd/issues/2884
 configure.args-append \
+    -DPCRE_ROOT_DIR=/usr/lib \
+    -DENABLE_LUAJIT=ON \
+    -DENABLE_TORCH=ON \
+    -DENABLE_SNOWBALL=ON \
+    -DENABLE_FANN=ON \
+    -DENABLE_GD=ON \
+    -DENABLE_LIBUNWIND=ON \
```

Comment:
This unwind option is causing build failures. See the azure build logs. e.g.

```
CMake Error at CMakeLists.txt:415 (MESSAGE):
  Cannot find library unwind for package LIBUNWIND
```

possibly you need to add `libunwind-headers` port as a build dep.

Reply:
Likewise, what have you changed here to resolve this ?

Reply:
Thanks. I added `port:unwind` under `depends_run`. Everything will build without, but the binary won't run without it.